ibis/formats/pandas.py

```diff
@@ -109,14 +109,11 @@ def infer_table(cls, df):
 
     @classmethod
     def convert_table(cls, df, schema):
-        if len(schema) != len(df.columns):
-            raise ValueError(
-                "schema column count does not match input data column count"
-            )
+        if schema.names != tuple(df.columns):
+            raise ValueError("schema names don't match input data columns")
```

Comment:
Previously we'd implicitly rename in `PandasData.convert_table`, but not in the other formats. This would lead to name mismatch bugs that would only show up in `to_pyarrow`/`to_polars` (hence the `Table.unnest()` bug noticed above that failed only in a doctest (goes through `to_pyarrow()`)). I've instead opted to remove this in `to_pandas` in favor of fixes upstream in the specific backends.

Reply:
Seems like the right move.